ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/ExecutionEngineJsonRpcMethod.java

```diff
@@ -45,18 +48,36 @@ public enum EngineStatus {
   private static final Logger LOG = LoggerFactory.getLogger(ExecutionEngineJsonRpcMethod.class);
   protected final Optional<MergeContext> mergeContextOptional;
   protected final Supplier<MergeContext> mergeContext;
+  protected final Optional<ProtocolSchedule> protocolSchedule;
   protected final ProtocolContext protocolContext;
   protected final EngineCallListener engineCallListener;
+  protected final Optional<ScheduledProtocolSpec.Hardfork> cancun;
```

Comment:
this is a V3 concern right?

Reply:
Yes, but used across 3 different apis, and will likely be used by more hardforks in the future.